File: Api/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Api.ViewModels;

namespace Api.Models
{
    public class Order : Auditable
    {
        public int Id { get; set; }
        
        public DateTime Date { get; set; }
        
        public int UserId { get; set; }
        public User? User { get; set; }
        
        public int AddressId { get; set; }
        public Address? Address { get; set; }
        
        public int StatusId { get; set; }
        public Status? Status { get; set; }
        
        public virtual List<ArticleOrder>? ArticleOrders { get; set; }
    }
}

File: WebApp/src/app/models/order.ts
import {Address} from "./address";
import {User} from "./user";
import {Status} from "./status";
import {Auditable} from "./auditable";
import {ArticleOrder} from "./article";

export interface Order extends Auditable{
     id?: number;
     date?: Date;
     userId?: number;
     user?: User;
     addressId?: number;
     address?: Address;
     statusId?: number;
     status?: Status;
     articleOrders?: Array<ArticleOrder>;
}
